pkg/osinfo/osinfo_darwin.go

```diff
@@ -0,0 +1,47 @@
+package osinfo
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+var defaultReturnValue = OSInfo{
+	OS:       "darwin",
+	Version:  "unknown",
+	Platform: "unknown",
+}
+
+func GetOSInfo() OSInfo {
+	out, err := getInfo()
+	if err != nil {
+		return defaultReturnValue
+	}
+	for strings.Contains(out, brokenPipeOutput) {
+		out, err = getInfo()
+		time.Sleep(retryWaitTime)
+	}
+	if err != nil {
+		return defaultReturnValue
+	}
+	osStr := replaceLineTerminations(out)
+	osInfo := strings.Split(osStr, " ")
+	oss := OSInfo{
+		OS:       osInfo[0],
+		Version:  osInfo[1],
+		Platform: osInfo[2],
+	}
```

Comment:
Comment for all code instances - this could easily panic and requires handling in case slice len != 3

Reply:
Done